homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -340,6 +359,24 @@ def __init__(self, device, bridge, bridge_device):
         self._bridge_unique_id = serial_to_unique_id(bridge_device["serial"])
         if "serial" not in self._device:
             return
+
+        if "parent_device" in device:
```

Comment:
I think you can write this as

```python
        if (
            "parent_device" in device
            and device_info_by_device_id is not None
            and (
                parent_device_info := device_info_by_device_id.get(
                    device["parent_device"]
                )
            )
            is not None
        ):
```

Reply:
or if `parent_device_info` will never be undefined

```python
        if (
            "parent_device" in device
            and device_info_by_device_id is not None
            and (
                parent_device_info := device_info_by_device_id.get(
                    device["parent_device"]
                )
            )
        ):
```

Reply:
great! I like it.  I"ll make the change.